[
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/212658196",
    "html_url": "https://github.com/electron/electron/issues/5228#issuecomment-212658196",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/5228",
    "id": 212658196,
    "user": {
      "login": "deepak1556",
      "id": 964386,
      "avatar_url": "https://avatars2.githubusercontent.com/u/964386?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/deepak1556",
      "html_url": "https://github.com/deepak1556",
      "followers_url": "https://api.github.com/users/deepak1556/followers",
      "following_url": "https://api.github.com/users/deepak1556/following{/other_user}",
      "gists_url": "https://api.github.com/users/deepak1556/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/deepak1556/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/deepak1556/subscriptions",
      "organizations_url": "https://api.github.com/users/deepak1556/orgs",
      "repos_url": "https://api.github.com/users/deepak1556/repos",
      "events_url": "https://api.github.com/users/deepak1556/events{/privacy}",
      "received_events_url": "https://api.github.com/users/deepak1556/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2016-04-21T00:00:15Z",
    "updated_at": "2016-04-21T00:00:15Z",
    "body": "Could you provide a simple app demonstrating the issue ?\n"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/212997197",
    "html_url": "https://github.com/electron/electron/issues/5228#issuecomment-212997197",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/5228",
    "id": 212997197,
    "user": {
      "login": "ccnokes",
      "id": 1610343,
      "avatar_url": "https://avatars3.githubusercontent.com/u/1610343?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/ccnokes",
      "html_url": "https://github.com/ccnokes",
      "followers_url": "https://api.github.com/users/ccnokes/followers",
      "following_url": "https://api.github.com/users/ccnokes/following{/other_user}",
      "gists_url": "https://api.github.com/users/ccnokes/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/ccnokes/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/ccnokes/subscriptions",
      "organizations_url": "https://api.github.com/users/ccnokes/orgs",
      "repos_url": "https://api.github.com/users/ccnokes/repos",
      "events_url": "https://api.github.com/users/ccnokes/events{/privacy}",
      "received_events_url": "https://api.github.com/users/ccnokes/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2016-04-21T16:29:31Z",
    "updated_at": "2016-04-21T16:29:48Z",
    "body": "Yeah, here you go: https://github.com/ccnokes/electron-did-fail-load. \n\nI was able to verify that behavior in 0.36.9 (my version) and the latest 0.37.6. My sample app is on the latest. \n\nI based it on electron-boilerplate so it's got all of that project's commit history on it, but here's my change set: \nhttps://github.com/ccnokes/electron-did-fail-load/commit/ce7f3a6d18615cbbf20bc09132bc81f58b7f0496\n\nAfter `npm install`ing it, run `node server.js` to start a node server that will serve a gif to download. Then start the electron app with `npm run build && npm run start`. Click the \"download gif\" link to see the behavior I'm talking about. Electron should emit a `did-fail-load` event with a `-3` code, and then crash once the native save file dialog is closed. \n\nIf it's helpful, I can try to PR a failing unit test for this. I wish I could jump in and debug it more but I don't really know C++ or ObjC ðŸ˜¦ .\n"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/213021747",
    "html_url": "https://github.com/electron/electron/issues/5228#issuecomment-213021747",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/5228",
    "id": 213021747,
    "user": {
      "login": "deepak1556",
      "id": 964386,
      "avatar_url": "https://avatars2.githubusercontent.com/u/964386?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/deepak1556",
      "html_url": "https://github.com/deepak1556",
      "followers_url": "https://api.github.com/users/deepak1556/followers",
      "following_url": "https://api.github.com/users/deepak1556/following{/other_user}",
      "gists_url": "https://api.github.com/users/deepak1556/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/deepak1556/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/deepak1556/subscriptions",
      "organizations_url": "https://api.github.com/users/deepak1556/orgs",
      "repos_url": "https://api.github.com/users/deepak1556/repos",
      "events_url": "https://api.github.com/users/deepak1556/events{/privacy}",
      "received_events_url": "https://api.github.com/users/deepak1556/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2016-04-21T17:11:25Z",
    "updated_at": "2016-04-21T17:14:12Z",
    "body": "Thanks for the test case, the `did-fail-load` is just a duplicate fire of `did-fail-provisional-load`, i think we should separate `did-fail-provisional-load` and `did-fail-load` since its causing confusion @zcbenz ? The crash stack trace\n\n```\n#3 0x0000009109c9 content::WebContentsUserData<>::FromWebContents()\n#4 0x0000009610bd atom::AtomDownloadManagerDelegate::OnDownloadPathGenerated()\n#5 0x0000009626cb base::internal::RunnableAdapter<>::Run()\n#6 0x000000962550 _ZN4base8internal12InvokeHelperILb1EvNS0_15RunnableAdapterIMN4atom27AtomDownloadManagerDelegateEFvjRKNS_8CallbackIFvRKNS_8FilePathEN7content12DownloadItem17TargetDispositionENS9_18DownloadDangerTypeES8_EEES8_EEENS0_8TypeListIJRKNS_7WeakPtrIS4_EERKjSG_S8_EEEE8MakeItSoESJ_SO_SQ_SG_S8_\n#7 0x0000009624bd _ZN4base8internal7InvokerINS_13IndexSequenceIJLm0ELm1ELm2EEEENS0_9BindStateINS0_15RunnableAdapterIMN4atom27AtomDownloadManagerDelegateEFvjRKNS_8CallbackIFvRKNS_8FilePathEN7content12DownloadItem17TargetDispositionENSC_18DownloadDangerTypeESB_EEESB_EEEFvPS7_jSJ_SB_EJNS_7WeakPtrIS7_EEjSH_EEENS0_8TypeListIJNS0_12UnwrapTraitsISQ_EENST_IjEENST_ISH_EEEEENS0_12InvokeHelperILb1EvSM_NSS_IJRKSQ_RKjSJ_SB_EEEEEFvSB_EE3RunEPNS0_13BindStateBaseESB_\n#8 0x000000884c56 base::Callback<>::Run()\n#9 0x0000008956a8 _ZN4base8internal12InvokeHelperILb0EvNS_8CallbackIFvRKNS_8FilePathEEEENS0_8TypeListIJS5_EEEE8MakeItSoES7_S5_\n#10 0x000000895650 _ZN4base8internal7InvokerINS_13IndexSequenceIJLm0EEEENS0_9BindStateINS_8CallbackIFvRKNS_8FilePathEEEES9_JS6_EEENS0_8TypeListIJNS0_12UnwrapTraitsIS6_EEEEENS0_12InvokeHelperILb0EvSA_NSC_IJS8_EEEEEFvvEE3RunEPNS0_13BindStateBaseE\n#11 0x7f496b1c817f base::debug::TaskAnnotator::RunTask()\n#12 0x7f496b1e74cb base::MessageLoop::RunTask()\n#13 0x7f496b1e7b6b base::MessageLoop::DoWork()\n#14 0x7f496b1b954a <unknown>\n#15 0x7f4964522d87 g_main_context_dispatch\n#16 0x7f4964522fe0 <unknown>\n#17 0x7f496452308c g_main_context_iteration\n#18 0x7f496b1b93a6 base::MessagePumpGlib::Run()\n#19 0x7f496b200ae0 base::RunLoop::Run()\n#20 0x7f496d20c28b content::BrowserMainLoop::MainMessageLoopRun()\n#21 0x7f496d20c094 content::BrowserMainLoop::RunMainMessageLoopParts()\n#22 0x7f496d20f02d <unknown>\n#23 0x7f496d2076fc content::BrowserMain()\n#24 0x7f496d184df1 <unknown>\n#25 0x7f496d183b60 content::ContentMain()\n#26 0x0000007e7ac2 main\n#27 0x7f496269b710 __libc_start_main\n#28 0x0000007e7915 <unknown>\n  r8: 0000000000000000  r9: 0000010e6c2b5e38 r10: 0000010e6bc0fe60 r11: 00007f49627e7020\n r12: 0000010e6bc0d320 r13: 00007ffc58321530 r14: 00007f496b25e846 r15: 00007f496b28eeab\n  di: 0000000000000010  si: 0000000000ee007c  bp: 00007ffc58321070  bx: 00007ffc583213c8\n  dx: 000000000000009c  ax: 0000000000ee007c  cx: 0000000000000000  sp: 00007ffc58321058\n  ip: 00007f496b214510 efl: 0000000000010202 cgf: 0000000000000033 erf: 0000000000000004\n trp: 000000000000000e msk: 0000000000000000 cr2: 0000000000000028\n```\n"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/213185261",
    "html_url": "https://github.com/electron/electron/issues/5228#issuecomment-213185261",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/5228",
    "id": 213185261,
    "user": {
      "login": "zcbenz",
      "id": 639601,
      "avatar_url": "https://avatars3.githubusercontent.com/u/639601?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/zcbenz",
      "html_url": "https://github.com/zcbenz",
      "followers_url": "https://api.github.com/users/zcbenz/followers",
      "following_url": "https://api.github.com/users/zcbenz/following{/other_user}",
      "gists_url": "https://api.github.com/users/zcbenz/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/zcbenz/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/zcbenz/subscriptions",
      "organizations_url": "https://api.github.com/users/zcbenz/orgs",
      "repos_url": "https://api.github.com/users/zcbenz/repos",
      "events_url": "https://api.github.com/users/zcbenz/events{/privacy}",
      "received_events_url": "https://api.github.com/users/zcbenz/received_events",
      "type": "User",
      "site_admin": true
    },
    "created_at": "2016-04-22T00:53:01Z",
    "updated_at": "2016-04-22T00:57:51Z",
    "body": "> As a side note, I was calling loadURL in the did-fail-load handler which caused Electron to crash. I can provide the full dump for that if you'd like.\n\n<del>The crash had been fixed in 0.37.x.</del>\n\nEDIT: Sorry, it seems to be a new crash.\n\n> i think we should separate did-fail-provisional-load and did-fail-load since its causing confusion\n\nIt would be a breaking change, #1242.\n"
  },
  {
    "url": "https://api.github.com/repos/electron/electron/issues/comments/290021151",
    "html_url": "https://github.com/electron/electron/issues/5228#issuecomment-290021151",
    "issue_url": "https://api.github.com/repos/electron/electron/issues/5228",
    "id": 290021151,
    "user": {
      "login": "aight8",
      "id": 1138623,
      "avatar_url": "https://avatars3.githubusercontent.com/u/1138623?v=3",
      "gravatar_id": "",
      "url": "https://api.github.com/users/aight8",
      "html_url": "https://github.com/aight8",
      "followers_url": "https://api.github.com/users/aight8/followers",
      "following_url": "https://api.github.com/users/aight8/following{/other_user}",
      "gists_url": "https://api.github.com/users/aight8/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/aight8/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/aight8/subscriptions",
      "organizations_url": "https://api.github.com/users/aight8/orgs",
      "repos_url": "https://api.github.com/users/aight8/repos",
      "events_url": "https://api.github.com/users/aight8/events{/privacy}",
      "received_events_url": "https://api.github.com/users/aight8/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-03-29T08:32:30Z",
    "updated_at": "2017-03-29T08:32:30Z",
    "body": "If the url is a file download ONLY \"did-fail-provisional-load\" is emitted. \"did-fail-load\" NOT anymore!"
  }
]